ivy/functional/ivy/linear_algebra.py

```diff
@@ -1129,14 +1130,15 @@ def matmul(
     )
 
 
-@handle_exceptions
-@handle_backend_invalid
-@handle_nestable
-@handle_array_like_without_promotion
-@handle_out_argument
-@to_native_arrays_and_back
-@handle_array_function
 @handle_device
+@infer_dtype
```

Comment:
The order of the relevant decorators applied to the function should be the reverse of the order [here](https://github.com/unifyai/ivy/blob/36a36bf1da65fa6f6b1c4077b2a64d5e0904ac8a/ivy/func_wrapper.py#L17)

Reply:
I took the order from here: https://unify.ai/docs/ivy/overview/deep_dive/function_wrapping.html so I'm guessing that deep-dive might need an update. Either way- fixed!